File: app/src/main/java/ru/garretech/garred/doramatv/database/FavoritesDAO.java
package ru.garretech.garred.doramatv.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import ru.garretech.garred.doramatv.model.Favorites;

@Dao
public interface FavoritesDAO {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long addFavorites(Favorites favorites);

    @Query("SELECT * FROM favorites")
    List<Favorites> getAllFavorites();

    @Query("SELECT * FROM favorites WHERE id = :ids")
    Favorites getFavoriteByIndex(long ids);

    @Query("SELECT * FROM favorites WHERE movie_url = :URL")
    Favorites getFavoriteByURL(String URL);

    @Delete
    void deleteFavorites(Favorites favorites);
}

File: app/src/main/java/ru/garretech/garred/doramatv/database/FavoritesDAO.kt
package ru.garretech.garred.doramatv.database

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import ru.garretech.garred.doramatv.model.Favorites

@Dao
interface FavoritesDAO {

    @get:Query("SELECT * FROM favorites")
    val allFavorites: List<Favorites>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addFavorites(favorites: Favorites): Long

    @Query("SELECT * FROM favorites WHERE movie_url = :URL")
    fun getFavoriteByURL(URL: String): Favorites?

    @Delete
    fun deleteFavorites(favorites: Favorites)

    @Query("DELETE FROM favorites")
    fun clearFavorites()
}
